### PR TITLE
fix: use escape instead of allow list when launching through Windows cmd

### LINKED ISSUE #35: Fails when editor is installed in directory containing spaces on windows
My vscode is installed in ` C:\Users\Isabella Skořepová\AppData\Local\Programs\Microsoft VS Code\Code.exe`, which is the default installation location. The first space is mostly my fault created when I was installing the system, but the other two are added by vscode installer.

It fails with error
```
'C:\Users\Isabella' is not recognized as an internal or external command, operable program or batch file.
```

Sidenote: I am using vite, but I see the same issue when trying to import the middleware directly in a vite.config.ts file.

### LINKED ISSUE #76: Fix for handling paths with square brackets on Windows
Fix for handling paths with square brackets on Windows
Path as pages\contest\[contest]\index.vue supported in windows.

Pull request for fix it https://github.com/yyx990803/launch-editor/pull/75

### PR DESCRIPTION
**Pull request by haoqunjiang**
Fixes #35
Fixes #76
Closes #54
Closes #57
Fixes #74#issuecomment-2269583713

### CODE REVIEW

**Comment by haoqunjiang** [comment]
@dominikg @btea I finally got the time to research the cumbersome and fragmented cmd documentation online and came up with this solution.
Could you help review it?

File: packages/launch-editor/index.js [hunk 1]
```diff
@@ -142,13 +112,55 @@ function launchEditor (file, specifiedEditor, onErrorCallback) {
...
+      return cmdArgs.replace(/([&|<>,;=^])/g, '^$1')
+    }
+
+    // Need to double quote the editor path in case it contains spaces;
+    // If the fileName contains spaces, we also need to double quote it in the arguments
+    // However, there's a case that it's concatenated with line number and column number
+    // which is separated by `:`. We need to double quote the whole string in this case.
+    // Also, if the string contains the escape character `^`, it needs to be quoted, too.
+    function doubleQuoteIfNeeded(str) {
+      if (str.includes(' ')) {
```
Comment (dominikg) [inline packages/launch-editor/index.js:147]: what if str has both space and ^ ?

File: packages/launch-editor/index.js [hunk 2]
```diff
@@ -142,13 +112,55 @@ function launchEditor (file, specifiedEditor, onErrorCallback) {
...
+        return `"${str}"`
+      } else if (str.includes('^')) {
+        // If a string includes an escaped character, not only does it need to be quoted,
+        // but the quotes need to be escaped too.
+        return `^"${str}^"`
+      }
+      return str
+    }
```
Comment (dominikg) [inline packages/launch-editor/index.js:155]:
```suggestion
    function doubleQuoteIfNeeded(str) {
      if (str.includes('^')) {
        // If a string includes an escaped character, not only does it need to be quoted,
        // but the quotes need to be escaped too.
        return `^"${str}^"`
      } else if (str.includes(' ')) {
        return `"${str}"`
      } 
      return str
    }
```

Comment (haoqunjiang) [reply packages/launch-editor/index.js]: Oops, good catch. Thanks!

**Review by dominikg — APPROVED** [review]

**Comment by dominikg** [comment]
approving based on documentation available, havn't tested it on an acutal windows machine due to lack thereof

**Review by btea — APPROVED** [review]
I did a quick test on Windows and it seems to be fine.

**Pull request merged**